[runtime] The return value from mono_class_from_mono_type must be released.

rolfbjarne opened the pull request:
Before:

    There were 258042 MonoObjects created, 235166 MonoObjects freed, so 22876 were not freed. (dynamic registrar)
    There were 205804 MonoObjects created, 204219 MonoObjects freed, so 1585 were not freed. (static registrar)

After:

    There were 258050 MonoObjects created, 235177 MonoObjects freed, so 22873 were not freed. (dynamic registrar)
    There were 205804 MonoObjects created, 204219 MonoObjects freed, so 1585 were not freed. (static registrar)

vs-mobiletools-engineering-service2 commented:
### :fire: Tests failed catastrophically on Build (no summary found). :fire:

Result file $(TEST_SUMMARY_PATH) not found. 
* [Azure DevOps](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=4812697&view=ms.vss-test-web.test-result-details)
* [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.11/4812697/sim/;/tests/vsdrops_index.html)
# GitHub pages

Results can be found in the following github pages (it might take some time to publish):

* [Test results](https://xamarin.github.io/macios.ci/pr/PR11729/4812697/HtmlReport-sim/tests/vsdrops_index.html)
* [API diff ](https://xamarin.github.io/macios.ci/pr/PR11729/4812697/HtmlReport-sim/api-diff/api-diff.html)
* [API & Generator diff](https://xamarin.github.io/macios.ci/pr/PR11729/4812697/apicomparison/api-diff.html)


[Pipeline](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=4812697&view=ms.vss-test-web.test-result-details) on Agent 
Merge e5acea56b317f73720fe9897484b1594fe2723ad into 165941ec143c8e71308a95e26e2de0fc0af4f77d

rolfbjarne commented:
/azp run

vs-mobiletools-engineering-service2 commented:
### :x: [PR Build] Tests failed on Build :x:

Tests failed on Build. 
* [Azure DevOps](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=4812974&view=ms.vss-test-web.test-result-details)
* [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.17/4812974/sim/;/tests/vsdrops_index.html)
# API diff
:white_check_mark: API Diff from stable
<details><summary>View API diff</summary>

* iOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.17/4812974/apidiff/;/ios-api-diff.html) [gist](https://gist.github.com/5f9d1a61d2431c27286742c57c8937eb)
* macOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.17/4812974/apidiff/;/mac-api-diff.html) [gist](https://gist.github.com/0168736da621e01ba54ea48dcb146f37)
* macCat [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.17/4812974/apidiff/;/maccat-api-diff.html) [gist](https://gist.github.com/9fd0c56b44c898764da1465520f30491)
* macCatiOS  [gist](https://gist.github.com/38beb425f20acc5c669dc1c389ada8ce)
* tvOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.17/4812974/apidiff/;/tvos-api-diff.html) [gist](https://gist.github.com/2c261834f400ac8800bacac8ef968465)
* watchOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.17/4812974/apidiff/;/watchos-api-diff.html) [gist](https://gist.github.com/0e6794181ef620d818907f4c20635ef8)
</details>

# API & Generator diff

:white_check_mark: [API Diff (from PR only)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.17/4812974/apigeneratordiff/;//api-diff.html) (no change)
:white_check_mark: [Generator Diff](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.17/4812974/apigeneratordiff/;//generator-diff/index.html) (no change)

# GitHub pages

Results can be found in the following github pages (it might take some time to publish):

* [Test results](https://xamarin.github.io/macios.ci/pr/PR11729/4812974/HtmlReport-sim/tests/vsdrops_index.html)
* [API diff ](https://xamarin.github.io/macios.ci/pr/PR11729/4812974/HtmlReport-sim/api-diff/api-diff.html)
* [API & Generator diff](https://xamarin.github.io/macios.ci/pr/PR11729/4812974/apicomparison/api-diff.html)


# Test results

<details>
<summary>2 tests failed, 106 tests passed.</summary>

## Failed tests

 * monotouch-test/Mac [dotnet]/Debug (CoreCLR) [dotnet]: Failed (Test run failed.
Tests run: 2444 Passed: 2353 Inconclusive: 7 Failed: 4 Ignored: 87)
 * monotouch-test/Mac [dotnet]/Debug (CoreCLR, static registrar) [dotnet]: Failed (Test run failed.
Tests run: 2441 Passed: 2353 Inconclusive: 7 Failed: 2 Ignored: 86)
</details>


[Pipeline](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=4812974&view=ms.vss-test-web.test-result-details) on Agent XAMBOT-1104.BigSur'
Merge e5acea56b317f73720fe9897484b1594fe2723ad into 51c936457d6ce0f7f596179f0fbe6b294c56f6c7

rolfbjarne commented:
Test failures are unrelated

* monotouch-test/Mac [dotnet]/Debug (CoreCLR) [dotnet]: Fixed in #11732, #11733 and #11734.
* monotouch-test/Mac [dotnet]/Debug (CoreCLR, static registrar) [dotnet]: Fixed in #11732 and #11733.
